sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java

```diff
@@ -39,14 +39,7 @@ public FunctionAppsImpl(final AppServiceManager manager) {
 
     @Override
     public FunctionApp getByResourceGroup(String groupName, String name) {
-        SiteInner siteInner = this.inner().getByResourceGroup(groupName, name);
-        if (siteInner == null) {
-            return null;
-        }
-        return wrapModel(
-            siteInner,
-            this.inner().getConfiguration(groupName, name),
-            this.inner().getDiagnosticLogsConfiguration(groupName, name));
+        return getByResourceGroupAsync(groupName, name).block();
```

Comment:
`this.inner().getByResourceGroup(groupName, name)` will cause backend exception if not found. So siteInner in the method will never be null.